src/ts/components/adskipbutton.ts

```diff
@@ -44,13 +44,14 @@ export class AdSkipButton extends Button<AdSkipButtonConfig> {
 
     let updateSkipMessageHandler = () => {
       // Display this button only if ad is skippable
-      // non skippable ads will return -1 for skippableAfter
-      if (skipOffset >= 0) {
-        this.show();
-      } else {
+      // non skippable ads will return -1 for skippableAfter for player version < v8.3.0
+      if (!skipOffset || skipOffset < 0) {
         this.hide();
+        return;
       }
 
+      this.show();
```

Comment:
Wouldn't it be cleaner to prevent the handler attachment inside the `adStartHandler` below? Now this handler is still unnecessarily executed on every `TimeChanged` event.